lib/sync.js

```diff
@@ -64,6 +64,11 @@ module.exports = function (x, options) {
                     pkg = opts.packageFilter(pkg, x);
                 }
 
+                if (pkg.main === '.') {
+                    var i = loadAsFileSync(path.resolve(x, '/index'));
+                    if (i) { return i; }
+                }
+
                 if (pkg.main) {
                     var m = loadAsFileSync(path.resolve(x, pkg.main));
```

Comment:
i feel like `pkg.main === '.' ? './index' : pkg.main` here might be simpler than the separate `if` statement - also, what happens if `.` is a directory and not a file? ie, let's say there's `./index/index.js` for some reason?

Maybe inside this if, we should `var main = pkg.main === '.' ? './index' : pkg.main;` and use `main` on both the m and n lines?